javalib/src/main/scala/java/util/AbstractMap.scala

```diff
@@ -0,0 +1,209 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+object AbstractMap {
+
+  private def equalsEntry[K, V](entry: Map.Entry[K, V], other: Any): Boolean = {
+    other match {
+      case entry2: Map.Entry[_, _] =>
+        (
+          (entry.getKey == null && entry2.getKey == null) ||
+          entry.getKey.equals(entry2.getKey)
+        ) && (
+            (entry.getValue == null && entry2.getValue == null) ||
+            entry.getValue.equals(entry2.getValue)
+          )
+      case _ => false
+    }
+  }
+
+  private def calcHashCode[K, V](entry: Map.Entry[K, V]): Int = {
+    val keyHash =
+      if (entry.getKey == null) 0
+      else entry.getKey.hashCode
+    val valueHash =
+      if (entry.getValue == null) 0
+      else entry.getValue.hashCode
+
+    keyHash ^ valueHash
+  }
+
+  class SimpleEntry[K, V](key: K, value: V) extends Map.Entry[K, V] with Serializable {
+    private var _key: K = key
+    private var _value: V = value
```

Comment:
You can simply put `private var _key: K, private var _value: V` in the constructor.